internal/resourcepolicies/resource_policies_mgr.go

```diff
@@ -0,0 +1,139 @@
+package resourcepolicies
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kbClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// resPoliciesMeta stored one resource policies and related resource policies configmap creation timestamp
+type resPoliciesMeta struct {
+	policies                *ResourcePolicies
+	configCreationTimestamp metav1.Time
+}
+
+// resPoliciesKey identify one specific resPoliciesMeta with refName and refType
+type resPoliciesKey struct {
+	refName string
+	refType string
+}
+
+// ResPoliciesManager manage all resource policies in memory and keep all up-to-date
+type ResPoliciesManager struct {
+	resPolicies map[resPoliciesKey]*resPoliciesMeta
+	namespace   string
+	kbClient    kbClient.Client
+}
+
+var ResPoliciesMgr ResPoliciesManager
+
+func (mgr *ResPoliciesManager) InitResPoliciesMgr(namespace string, kbClient kbClient.Client) {
+	mgr.resPolicies = make(map[resPoliciesKey]*resPoliciesMeta)
+	mgr.namespace = namespace
+	mgr.kbClient = kbClient
+}
+
+// getResourcePolicies first check and update current resource policies and then get the latest resource policies
+func (mgr *ResPoliciesManager) getResourcePolicies(refName, refType string) (*ResourcePolicies, error) {
+	if err := mgr.updateResourcePolicies(refName, refType); err != nil {
+		return nil, errors.Wrap(err, "failed to update resource policies meta info")
+	}
+
+	resKey := resPoliciesKey{
+		refName: refName,
+		refType: refType,
+	}
+	if policies, exist := mgr.resPolicies[resKey]; exist {
+		return policies.policies, nil
+	}
+	return nil, fmt.Errorf("failed to get resource policies")
+}
+
+func (mgr *ResPoliciesManager) updateResourcePolicies(refName, refType string) error {
+	// TODO currently we only want to support type of configmap
+	if refType != "configmap" {
+		return fmt.Errorf("unsupported resource policies reference type %s", refType)
+	}
+
+	resKey := resPoliciesKey{
+		refName: refName,
+		refType: refType,
+	}
+	policiesConfigmap := &v1.ConfigMap{}
+	err := mgr.kbClient.Get(context.Background(), kbClient.ObjectKey{Namespace: mgr.namespace, Name: refName}, policiesConfigmap)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get resource policies %s/%s configmap", refName, mgr.namespace)
+	}
+	if policiesConfigmap == nil {
+		return errors.Wrapf(err, "get empty resource policies %s/%s configmap", refName, mgr.namespace)
+	}
+
+	updateResPolicies := func() error {
```

Comment:
The name is `updateResPolicies`, but it doesn't update the policy, it creates a policy